litestar/plugins.py

```diff
@@ -107,11 +103,13 @@ def is_plugin_supported_type(value: Any) -> TypeGuard[ModelT]:
         """
         raise NotImplementedError()
 
-    def to_openapi_schema(self, model_class: type[ModelT]) -> Schema:
-        """Given a model class, transform it into an OpenAPI schema class.
+    def to_openapi_schema(self, annotation: Any, schema_creator: SchemaCreator, dto_for: ForType | None) -> Schema:
```

Comment:
this is a breaking change - reason for this is two-fold:

1. that we cannot actually create schemas only based on the annotation only, we need the schema_creator and dto_for info. 
2. the assumption of the plugin was that we are dealing with "models" - but pydantic actually has to serialize a lot of specialized annotations that are not models. Hence the plugin needs to be more generalized.

Reply:
note - schema_creator is currently in a private namespace `_openapi`. I dont actually want to expose it to users TBH, but there is little we can do given we decided on this switch. We might need to document it. Thoughts?